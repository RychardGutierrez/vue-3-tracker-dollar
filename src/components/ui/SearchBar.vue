<template>
  <div class="pt-16 mb-8 relative">
    <!-- search country -->
    <input v-model="searchQuery" @input="getResults" type="text" placeholder="Search for country"
      class="py-2 px-1 w-full bg-transparent border-b border-y-slate-400 focus:border-dollar-primary focus:outline-none focus:shadow-[0px_1px_0_0_#004E71]" />

    <!-- Slot- you custom list result -->
    <slot :countrySearchResults="countrySearchResults" :searchError="searchError" />
  </div>
</template>

<script setup>

import useSearchCountry from '@/composables/useSearchCountry';

const { getResults, searchQuery, searchError, countrySearchResults } = useSearchCountry();

</script>

<style lang="scss" scoped>

</style>