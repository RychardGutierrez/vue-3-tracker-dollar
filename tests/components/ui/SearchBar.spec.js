import { expect } from '@jest/globals';
import { flushPromises, shallowMount } from '@vue/test-utils';

import SearchBar from '../../../src/components/ui/SearchBar.vue';

const mockUseRouteCountry = {
  getResults: jest.fn(),
  searchQuery: '',
  searchError: { error: false, message: '' },
  countrySearchResults: null,
};

jest.mock('../../../src/composables/useSearchCountry', () => {
  const useRouteCountry = () => mockUseRouteCountry;
  return useRouteCountry;
});

describe('SearchBar composables', () => {
  let wraper = null;
  beforeEach(() => {
    wraper = shallowMount(SearchBar);
  });

  afterEach(() => {
    jest.clearAllMocks();
  });

  test('should to match with snapshot ', () => {
    expect(wraper.html()).toMatchSnapshot();
  });

  test('should get result when input have some text', async () => {
    wraper = shallowMount(SearchBar);
    const inputSearch = wraper.find('input');
    await inputSearch.setValue('Bolivia');
    await flushPromises();

    expect(mockUseRouteCountry.getResults).toHaveBeenCalledTimes(1);
  });
});
