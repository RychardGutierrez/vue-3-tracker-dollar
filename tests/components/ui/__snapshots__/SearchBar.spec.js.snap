// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar composables should to match with snapshot  1`] = `
"<div class=\\"pt-16 mb-8 relative\\">
  <!-- search country --><input type=\\"text\\" placeholder=\\"Search for country\\" class=\\"py-2 px-1 w-full bg-transparent border-b border-y-slate-400 focus:border-dollar-primary focus:outline-none focus:shadow-[0px_1px_0_0_#004E71]\\"><!-- Slot- you custom list result -->
</div>"
`;
